rosidl_runtime_rs/src/traits.rs

```diff
@@ -159,3 +159,95 @@ pub trait Service: 'static {
     /// Get a pointer to the correct `rosidl_service_type_support_t` structure.
     fn get_type_support() -> *const std::os::raw::c_void;
 }
+
+/// Trait for actions.
+///
+/// User code never needs to call this trait's method, much less implement this trait.
+pub trait Action: 'static {
+    /// The goal message associated with this action.
+    type Goal: Message;
+
+    /// The result message associated with this action.
+    type Result: Message;
+
+    /// The feedback message associated with this action.
+    type Feedback: Message;
+
+    /// Get a pointer to the correct `rosidl_action_type_support_t` structure.
+    fn get_type_support() -> *const std::os::raw::c_void;
```

Comment:
nit: This can be done in another PR because this is not the only place in ros2_rust :sweat_smile: but this `std::os::raw::c_void` type is an alias to [std::ffi::c_void](https://doc.rust-lang.org/std/ffi/enum.c_void.html) which is the more appropriate type to use. 

`std::os::raw::c_void` only exists to maintain backwards compatibility with the Rust compiler <1.30 (until 1.1). Here is the RFC where this was introduced, [RFC 2521](https://github.com/rust-lang/rfcs/blob/master/text/2521-c_void-reunification.md).

Reply:
Sure, I've updated all the uses in `rosidl_runtime_rs` and `rosidl_generator_rs` to use `std::ffi::c_void` instead. I'll leave `rclrs` as-is for this PR.